src/main/java/org/cloudfoundry/promregator/config/Target.java

```diff
@@ -145,7 +146,7 @@ public void setAuthenticatorId(String authenticatorId) {
 	 */
 	public List<String> getPreferredRouteRegex() {
 		if (this.preferredRouteRegex == null) {
-			return null;
+			return Collections.emptyList();
```

Comment:
Nope. That's a change of semantics: An empty list of PreferredRoutRegex is something different than none of them (`null`). See also https://github.com/promregator/promregator/blob/94688a4530a6d58d12e4adf74c21ab68c3b31a82/src/main/java/org/cloudfoundry/promregator/config/Target.java#L168

PS: I thought that @checketts already had had a similar suggestion before, which I rejected. However, I couldn't find it anymore.

Reply:
Nope. I don't recall making any proposal around that.

Reply:
Strange 😕 thought so... but doesn't matter... Thanks for the pointer, though!

Reply:
I understand that it is a change of semantics, however I traced it through the codepath as to where it would be used, in ReactiveAppInstanceScanner.javam line +/- 322

`if (patterns == null || patterns.isEmpty()) {
			log.debug("No Preferred Route URL (Regex) provided; taking first Application Route in the list provided");
			return urls.get(0);
		}`

And the code treats empty list and null the same..

Reply:
I get your point. However, then we should make this consistent also into the other direction:
* Declare the return of `getPreferredRouteRegexPatterns` and `getPreferredRouteRegex` to be `@NonNull`
* Document that briefly in a JavaDoc comment.
* Remove the `patterns == null` comparison at https://github.com/promregator/promregator/blob/58a69f9ecd1f0a15310086a0d3bc5ed6cdc54a0e/src/main/java/org/cloudfoundry/promregator/scanner/ReactiveAppInstanceScanner.java#L322  as they are redundant.
* If we find further spots like this (where we are checking something on `null` which can't be `null`, we should get rid of them as well.

What do you think?

Reply:
I wholeheartedly concur